adapter/internal/discovery/xds/rate_limiter_cache.go

```diff
@@ -357,6 +357,24 @@ func AddSubscriptionLevelRateLimitPolicy(policyList *types.SubscriptionPolicyLis
 			rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType][policy.Organization] = make(map[string]*rls_config.RateLimitDescriptor)
 		}
 		rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType][policy.Organization][policy.Name] = descriptor
+
+		if policy.RateLimitCount > 0 && policy.RateLimitTimeUnit != "" {
+			burstCtrlUnit, err := parseRateLimitUnitFromSubscriptionPolicy(policy.RateLimitTimeUnit)
+			if err != nil {
+				loggers.LoggerXds.Error("Error while getting the burst control time unit", err)
+			}
+			burstCtrlPolicyConfig := rls_config.RateLimitPolicy{
+				Unit:            burstCtrlUnit,
+				RequestsPerUnit: uint32(policy.RateLimitCount),
+			}
+			burstCtrlPolicyName := "burst"+ policy.Name
+			burstCtrlDescriptor := &rls_config.RateLimitDescriptor{
+				Key: "policy",
+				Value: burstCtrlPolicyName,
+				RateLimit: &burstCtrlPolicyConfig,
+			}
+			rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType][policy.Organization][burstCtrlPolicyName] = burstCtrlDescriptor
```

Comment:
If we add a burst control unit at the same level where the default subscription policy exists there could be conflicts.
Eg:

1. Create a policy named "Gold"
2. Create another policy named "burstGold"

This will lead to a conflict.

Let's move burst control inside the policy.

Suggested YAML:

```yaml
- Key: Org
  Value: 1234
  RateLimitDescriptor:
    - Key: Policy
      Value: Gold
      RateLimit:
        Unit: hour
        RequestsPerUnit: 200000
      RateLimitDescriptor:
        - Key: Burst
          Value: Enabled
          RateLimit:
            Unit: min
            RequestsPerUnit: 100
```

Reply:
Why we need to assign a value for subscription  and policy here ?

Reply:
Burst control is under a subscription policy right? Why do you think we don't need it?

Reply:
We don't configure rate-limiter with values for above mentioned fields.